### PR TITLE
virtual_server_compile_sections() The offset could be 0 too

### PR DESCRIPTION
**Pull request by jpereira**

### CODE REVIEW

File: src/lib/server/virtual_servers.c [hunk 1]
```diff
@@ -1301,10 +1301,10 @@ int virtual_server_compile_sections(CONF_SECTION *server, virtual_server_compile
 			 *	Cache the CONF_SECTION which was found.
 			 */
 			if (uctx) {
-				if (list[i].offset > 0) {
+				if (list[i].offset >= 0) {
```
Comment (arr2036) [inline src/lib/server/virtual_servers.c:1304]: Useless comparison covers the entire range.

File: src/lib/server/virtual_servers.c [hunk 2]
```diff
@@ -1301,10 +1301,10 @@ int virtual_server_compile_sections(CONF_SECTION *server, virtual_server_compile
...
 					*(CONF_SECTION **) (((uint8_t *) uctx) + list[i].offset) = subcs;
 				}
-				if (list[i].instruction > 0) {
+				if (list[i].instruction >= 0) {
```
Comment (arr2036) [inline src/lib/server/virtual_servers.c:1307]: Useless comparison covers the entire range.

Comment (arr2036) [reply src/lib/server/virtual_servers.c:1307]: Trying out my new role as static analysis tool

**Comment by alandekok** [comment]
If `offset` isn't set, it's initialized to zero.  So we can't check for that.  It's better just to pad the input structs.

**Pull request closed without merging**